src/plugins/dashboard/public/dashboard_app/dashboard_app.tsx

```diff
@@ -188,6 +190,24 @@ export function DashboardApp({
     getScreenshotContext,
   ]);
 
+  useMemo(() => {
+    if (!dashboardAPI) return;
+    dashboardAPI?.expandedPanelId.subscribe(() => {
```

Comment:
Looks like we aren't unsubscribing from `expandedPanelId` on unmount 🤔

Reply:
Ah! Good catch - let me know if I'm misunderstanding in adding the unsubscribe in the useEffect cleanup on line 122 thanks!

Reply:
Hmmm... I'm honestly not sure what calling `unsubscribe` on the `BehaviourSubject` even does? I looked through the docs, and couldn't find anything. In this case, the pattern we use is to unsubscribe **from the Subscription object** that is returned when you subscribe - if you switch the subscription to a `useEffect` as mentioned in https://github.com/elastic/kibana/pull/190086#discussion_r1745982072, this would be easier :)

LMK if that doesn't make sense.

Reply:
Its important to call unsubscribe to clean up subscription. Otherwise its a memory leak

Reply:
> Unsubscribes removes the subscription from the behavior subject.

Haha, I know that! :) I am specifically asking about the difference between the `unsubscribe` from the `BehaviourSubject` object (i.e. calling `dashboardApi.someSubject.unsubscribe()` and the `unsubscribe` from the `Subscription` object (i.e. calling `someSubscription.unsubscribe()`. We do the latter (and I understand what that does), but @rshen91 was trying to do the first (which is what I'm unsure about). 

But that's okay - this is a minor thing 😆 Either way @rshen91, we should unsubscribe from the **subscription** and not the behaviour subject :+1: Sorry about the noise

Reply:
I think Im following 🤔 [c8cfcdd](https://github.com/elastic/kibana/pull/190086/commits/c8cfcdd10bd7c3b0730b1ffc61a2215532420b00) for changes.